### PR TITLE
Add missing features

### PR DESCRIPTION
**Pull request by hicom150**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[x] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
I've added the missing features one by commit

- Add better unit test (I've also added few integration tests but they are skipped)
- Add IPv6 support
- Add flags support
- Add message rate check

**Which issue (if any) does this pull request address?**

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: index.js
Comment (KayleePop) [inline index.js:187]: seems like these could be more DRY

**Review by KayleePop — APPROVED** [review]
didn't do a super thorough review, so please check over everything once more before merging, but LGTM

File: README.md
Comment (KayleePop) [inline README.md:132]: should we make an API for the accepted flags? Like we could use function arguments like `supports_utp` `prefers_encryption` etc and then set them as properties on the peer object like `peer.supports_utp === true`

we should decide now before cementing an API

Comment (KayleePop) [reply README.md:132]: actually, not sure if I fully understand the BEP, but flags should be a list not just one number I think?

File: index.js
Comment (KayleePop) [inline index.js:200]: maybe this would be simpler if there were a separate `_localAddedPeers` for ipv6

File: README.md
Comment (hicom150) [reply README.md:132]: Yeah! I think it is a good idea handle the flags as properties 👍 I'll try to work on it a bit 😉

File: index.js
Comment (hicom150) [reply index.js:200]: The good part of having it in the same object is that when we have to choose only 50 peers, there are in the same "bucket" so you don't have to be taking them from different places 😅
Giving that the filtering seems quite cheep, I seen more "pros" than "cons". What do you think?

**Comment by hicom150** [comment]
As always, great feedback @KayleePop 👍
I'll try to work in flags, I'll let you know when It's ready 😉

File: index.js
Comment (KayleePop) [inline index.js:226]: should this be 

```js
.map(flagInt => Buffer.from([flagInt]))
.reduce((acc, cur) => Buffer.concat([acc, cur]), Buffer.alloc(0))
```

? (and for `add6Flags` too)

File: index.js
Comment (KayleePop) [reply index.js:200]: hm, yea

It might be more readable if you split the localAdded and localDropped into ipv4 and ipv6 local variables though, instead of filtering multiple times below. Not sure. It's just kind of hard to parse

File: index.js
Comment (hicom150) [reply index.js:176]: Addressed that in the last refactor 👍

File: README.md
Comment (hicom150) [reply README.md]: I've added an API for flags, hope it is more clear now 😉

File: index.js
Comment (hicom150) [reply index.js:219]: I've refactored this part, hope it is more clear now 😅

File: index.js
Comment (hicom150) [reply index.js]: Nice catch! This was actually a bug, so I fixed and added more test 👍

**Comment by hicom150** [comment]
@KayleePop I've added two commits **Add better flags API** and **Refactor with review suggestions** to address your suggestions.
I'd appreciate if you could give it a last review just to check if everything fits 😅

Thank you! ❤️

**Comment by KayleePop** [comment]
great! looks good I think

**Comment by feross** [comment]
Nice work, @hicom150. This is really great. We finally addressed those TODOs which have been in the readme since the very earliest days!

The only last bit of feedback I have is that it might be nice to camelCase the properties in the flags object.

**Pull request merged**
